### DISCUSSION TITLE
Pointing a session to a specific node?

### DISCUSSION THREAD

**Discussion by georgiosd** [ravendb/ravendb#18989] [Q&A]
Hey all

I am running into an issue where, during the processing of a subscription, the document is not available if I re-read it with a cluster-wide session, presumably because the cluster-wide session reads from a different node than the batch session. 

Any ideas how to resolve this? Like maybe asking the cluster-wide session to read from the same node as the sub?

pseudo code:
```
OnProcessSubscriptionDocument(item) {
  var clusterWideSession = DocumentStore.OpenAsyncSession(ClusterWide);

  var doc = await clusterWideSession.LoadAsync(item.Id); // => null
}
```

**Comment by ayende** [comment]
In a subscription, you should use `batch.OpenSession` which will force it to use the same node as the subscription came from.
That will also load the items from the batch into the session, saving a trip.

Beyond that, you can't really force it to use a specific node (what happens if it is down?).

**Reply by georgiosd** [reply]
Correct me if I'm wrong, `batch.OpenSession` doesn't allow a cluster-wide session to be opened, and I need it here.
